client/upload/upload_client.go

```diff
@@ -108,3 +122,21 @@ func (u *Uploader) complete(objectKey string, uploadID *string, partNumbers []*s
 	}
 	return nil
 }
+
+func getFileSize(fd io.Reader) (int64, error) {
+	var length int64
+	switch r := fd.(type) {
+	case io.Seeker:
+		pos, _ := r.Seek(0, 1)
+		defer r.Seek(pos, 0)
+
+		n, err := r.Seek(0, 2)
+		if err != nil {
+			return length, err
+		}
+		length = n
+
```

Comment:
Remove this empty line.

Reply:
`getFileSize` will return `0` while `fd` is not a Seeker, is this expected?

Reply:
`getFileSize()` followed the way `initSize()` in `client/upload/chunk.go` . Both of them can get the file size when `io.Reader` is assigned in parameter. 

In other words, I think the `case io.Seeker` will always execute to get filesize.

Reply:
Maybe we should handle not seekable file like `os.Stdin`? At least, we should return an error while file not seekable.

Reply:
I think maybe`os.Stdin`  is not necessary and it makes `upload` complicated.So I add an check whether file is seekable.